File: Online_Judge_Problems/BOJ/boj_10870/boj_10870.py
'''---------------------------------------------
Sub  : [BOJ] 피보나치 수 5
Link : https://www.acmicpc.net/problem/10870
Level: Bronze 2
Tag  : Python, 피보나치, 동적계획
Memo
 - 
---------------------------------------------'''

import sys
input = sys.stdin.readline

n = int(input().strip())

a, b = 0, 0

for i in range(n):
    if(i==0):
        a, b = 0, 1
        continue
    a, b = b ,a+b

print(b)

File: Online_Judge_Problems/boj/boj_10870/boj_10870.js
/*----------------------------------------------
Sub  : [BOJ] 피보나치 수 5
Link : https://www.acmicpc.net/problem/10870
Level: 브론즈 2
Tag  : JS, 피보나치, 동적계획
Memo
----------------------------------------------*/

const fs = require('fs')
const input = fs.readFileSync(0, 'utf-8').trim().split('\n')

n = Number(input[0])

let [a, b] = [0, 0]

for (let i=0; i<n; i++) {
  if (i === 0){
    [a, b] = [0, 1]
    continue;
  }
  let temp  = b
  b = a+temp
  a = temp
}

console.log(b)
